lib/gather/collector.rb

```diff
@@ -40,18 +41,33 @@ def self.physical_data
       data[:model] = `dmidecode -s system-product-name`.chomp
       data[:bios] = `dmidecode -s bios-version`.chomp
       data[:serial] = `dmidecode -s system-serial-number`.chomp
-      data[:ram] = `grep MemTotal /proc/meminfo | awk '{print $2}'`.chomp # RAM measured in kB
+
+      # Get RAM info
+
+      data[:ram] = {}
+      ram_info = `dmidecode -q -t 17`.split('Memory Device')[1..]
+      data[:ram][:total_capacity] = si_to_bytes(between(`dmidecode -t 16`, 'Maximum Capacity: ', "\n")) / (1000.0 ** 3)
+      data[:ram][:units] = ram_info&.size
```

Comment:
This shows the number of _slots on the motherboard_ not the _number of dimms installed_. 

For example, a system with 24 slots but 16 x 8GB slots installed will report that it has 128GB RAM with 5.3GB per slot.

The more appropriate way of identifying the number of used slots is to either:
- Exclude the banks from the output that have `Size: No Module Installed`
- Count the used slots with the sys filesystem like `ls /sys/devices/system/edac/mc/mc*/ |grep dimm |wc -l`. Although this doesn't work for cloud/virtual systems and will return 0 so this data can be used and default to 1 for virtual (as virtual devices seem to always report memory as a single slot with the total) as follows: `(grep dimm <(ls /sys/devices/system/edac/mc/mc*/ 2> /dev/null ) || echo 'virtual') |wc -l`

Reply:
Fixed 84ded74